Assets/Scripts/Game/LevelingSystem.cs

```diff
@@ -0,0 +1,175 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class LevelingSystem : MonoBehaviour
+{
+    [Header("Experience")]
+    [SerializeField] private int xpPerLevelBase = 80;
+    [SerializeField] private int xpPerLevelGrowth = 30;
+
+    [Header("Leveling")]
+    [Tooltip("Set which properties should increase when the player levels up and by how much.")]
+    [SerializeField] private EntityProperty[] baseLevelUpBonuses;
+    [Tooltip("Set which properties should increase when the player picks Quackers and by how much.")]
+    [SerializeField] private EntityProperty[] quackersLevelUpBonuses;
+    [Tooltip("Set which properties should increase when the player picks Flappers and by how much.")]
+    [SerializeField] private EntityProperty[] flappersLevelUpBonuses;
+    [Tooltip("Set which properties should increase when the player picks Tappers and by how much.")]
+    [SerializeField] private EntityProperty[] tappersLevelUpBonuses;
+
+    public int XP
+    {
+        get { return GameManager.Player.GetProperty<int>(PropertyName.XP); }
+        private set { GameManager.Player.SetProperty<int>(PropertyName.XP, value); }
+    }
+
+    public int Level
+    {
+        get { return GameManager.Player.GetProperty<int>(PropertyName.Level); }
+        private set { GameManager.Player.SetProperty<int>(PropertyName.Level, value); }
+    }
+
+    public int LevelUps
+    {
+        get { return GameManager.Player.GetProperty<int>(PropertyName.LevelUps); }
+        private set { GameManager.Player.SetProperty<int>(PropertyName.LevelUps, value); }
+    }
+
+    public int Quackers
+    {
+        get { return GameManager.Player.GetProperty<int>(PropertyName.Quackers); }
+        private set { GameManager.Player.SetProperty<int>(PropertyName.Quackers, value); }
+    }
+
+    public int Flappers
+    {
+        get { return GameManager.Player.GetProperty<int>(PropertyName.Flappers); }
+        private set { GameManager.Player.SetProperty<int>(PropertyName.Flappers, value); }
+    }
+
+    public int Tappers
+    {
+        get { return GameManager.Player.GetProperty<int>(PropertyName.Tappers); }
+        private set { GameManager.Player.SetProperty<int>(PropertyName.Tappers, value); }
+    }
+
+    void Start()
+    {
+        XP = 0;
+        Level = ToLevel(XP);
+        LevelUps = 0;
+        Quackers = 0;
+        Flappers = 0;
+        Tappers = 0;
+    }
```

Comment:
I'm fairly sure this comment will be a low IQ remark, but once we have actual character data for these values and we want to save / load, will initializing these values to 0 muck it up? I imagine this is just a placeholder I just wanted to confirm?

Reply:
Yeah this is just a placeholder for a save/load function. Good catch!